Copy new lang assumption to all languages rules (#1162)
act-rules/act-rules.github.io: _rules/element-lang-valid-de46e4.md

## Assumptions

> The language of the page can be set by other methods than the `lang` attribute, for example using HTTP headers or the `meta` element. These methods are not supported by all assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.2: Language of Parts](https://www.w3.org/TR/WCAG21/#language-of-parts).

Reviewer: Is this relevant to language-of-parts? The other methods are all at the page level (HTTP headers, `meta` element in header, default browser language)

Reviewer: If there's no `lang` attribute on the element, then language of page is relevant since it affects the inherited language

Reviewer 2: Perhaps this version simply needs a short sentence at the start to be clear on that relevancy. I don't fully understand the inner browser workings around language determination, but maybe something like:

"If an element lang attribute does not validate, it will default to the language of the page."

Author: Yes. I was really not sure about the relevance of this version of the assumption for the "language of parts" rule, but decided to get the discussion going 😄

Reviewer 3: I agree this only makes sense if prefaced by what @EmmaJP said. Otherwise, it would not be clear how the assumption relates to "Language of Parts".

Reviewer: If the language is invalid the behaviour is pretty random:

- something it's the inherited lang
- something it's the page lang 
- something it's the system lang

Basically invalid lang means the screen reader picks a language at random (aka undefined behaviour). Could say something like:

"If an element lang attribute does not validate, behaviour is undefined, and content is read in a random language."

Reviewer 4: ```suggestion
The language of the page can be set by other methods than the `lang` attribute; using HTTP headers or the `meta` element. These methods are not known to be supported by assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.2: Language of Parts](https://www.w3.org/TR/WCAG21/#language-of-parts).
```

Author: There was already a new assumption for "language of element" worked in #1133 that was merged today. So I just kept it 😇
